### PR TITLE
PUBDEV-6129 Added support for using indices when specifying column-related parameters of target encoding(Python API)

### PR DESCRIPTION
**Pull request by deil87**
1) added support for using indices for column-related parameters
2) renamed parameter for blending ( deprecated previous one)

### CODE REVIEW

File: h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_automl_gbm_target_encoding.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def titanic_with_te_kfoldstrategy(frame = None, seeds = None):
 
       teColumns = ["home.dest", "cabin", "embarked"]
       targetEncoder = TargetEncoder(x= teColumns, y= targetColumnName,
-                                    fold_column= foldColumnName, blending_avg= True, inflection_point = 3, smoothing = 1)
+                                    fold_column= foldColumnName, blended_avg= True, inflection_point = 3, smoothing = 1)
```
Comment (ledell) [inline h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_automl_gbm_target_encoding.py]: Is this to match the R API?  Should this be in a separate PR?

**Comment by ledell** [comment]
@deil87 Just double checking -- you are using 0-index for Python, correct?  Also, is this already supported in R (if not, do you plan to add support?).

File: h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_automl_gbm_target_encoding.py (hunk 1 above)
Comment (deil87) [reply h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_automl_gbm_target_encoding.py]: @ledell the goal was to match the R API. Ideally it should be in a separate PR, yes. Is it ok to  leave it here?

**Comment by deil87** [comment]
@ledell Yes it is supported in Py and R, and I'm using 0-based indexing in both cases.

File: h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_automl_gbm_target_encoding.py (hunk 1 above)
Comment (ledell) [reply h2o-py/tests/testdir_algos/automl/target_encoding/pyunit_automl_gbm_target_encoding.py]: Yeah we can leave it here.  Just wanted to check.  Thanks.
